### ISSUE TITLE
Incompatible peer dependencies in a workspace with Next.js and Expo projects

### ISSUE THREAD

**Issue by ukch** [nrwl/nx#9846]
## Current Behavior
The latest version of `@nrwl/next` has a `peerDependency` on Next.js 12, which in turn has a peer dependency on React 17.0.2 or above. The new `@nrwl/expo` has a peer dependency on the latest 'expo', which has a peer dependency on react-native 0.64.3 which supports React 17.0.1 only. Using npm it seems impossible to have both types of projects in the same workspace without using legacy peer dependencies (which is risky and breaks stuff).

## Steps to Reproduce

* Create a new nx workspace
* `nx g @nrwl/next:app next-app`
* `nx g @nrwl/expo:app expo-app`
* `npm install`

### Failure Logs
```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: boilerplate-app@0.0.0
npm ERR! Found: react@17.0.1
npm ERR! node_modules/react
npm ERR!   react@"17.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^17.0.2 || ^18.0.0-0" from next@12.1.2
npm ERR! node_modules/next
npm ERR!   next@"12.1.2" from the root project
npm ERR!   peer next@">=10.2.0" from eslint-config-next@12.1.5
npm ERR!   node_modules/eslint-config-next
npm ERR!     dev eslint-config-next@"^12.1.5" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

### Environment
```
   Node : 14.19.1
   OS   : linux x64
   npm  : 8.1.0
   
   nx : 13.10.2
   @nrwl/angular : Not Found
   @nrwl/cypress : 13.10.2
   @nrwl/detox : 13.10.2
   @nrwl/devkit : 13.10.2
   @nrwl/eslint-plugin-nx : 13.10.2
   @nrwl/express : Not Found
   @nrwl/jest : 13.10.2
   @nrwl/js : 13.10.2
   @nrwl/linter : 13.10.2
   @nrwl/nest : Not Found
   @nrwl/next : 13.10.2
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 13.10.2
   @nrwl/react-native : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : 13.10.2
   @nrwl/web : 13.10.2
   @nrwl/workspace : 13.10.2
   typescript : 4.3.5
   rxjs : 6.6.7
   ---------------------------------------
   Community plugins:
   	 @nrwl/expo: 13.9.0
```

**Comment by muradbuyukasik** [comment]
As per the [React v18 release blog post](https://reactjs.org/blog/2022/03/29/react-v18.html), React v18 will ship in React Native with the New React Native Architecture. See [reactwg/react-native-new-architecture](https://github.com/reactwg/react-native-new-architecture). It seems like [opting-in](https://reactnative.dev/blog/2022/03/30/version-068#opting-in-to-the-new-architecture) is already possible, however we'll have to wait until Expo supports it as well.

**Comment by ukch** [comment]
FYI I think this might be fixable by upgrading expo to the new 45 SDK...

**Comment by xiongemi** [comment]
i think after you upgrade to latest @nrwl/expo version, it will be able to use react 18 with it.

**Comment by whoisdominic** [comment]
I am also seeing this issue in a fresh project. I can run `npm i --force` just fine but I don't think this is the proper solution.

I've aslo tried `@nrwl/expo@15.0.0-beta.4`

**Comment by zirho** [comment]
similar

@nrwl/react-native
@nrwl/react 

does not work well together

Let me try different version to match peer deps.

```
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR!
npm ERR! While resolving: react-native@0.70.5
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"18.2.0" from the root project
npm ERR!   peer react@"^18.0.0" from @testing-library/react@13.4.0
npm ERR!   node_modules/@testing-library/react
npm ERR!     dev @testing-library/react@"13.4.0" from the root project
npm ERR!   6 more (react-dom, react-router, react-router-dom, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"18.1.0" from react-native@0.70.5
npm ERR! node_modules/react-native
npm ERR!   peer react-native@"^0.70.1" from @nrwl/react-native@15.0.13
npm ERR!   node_modules/@nrwl/react-native
npm ERR!     dev @nrwl/react-native@"^15.0.13" from the root project
npm ERR!
npm ERR! Conflicting peer dependency: react@18.1.0
npm ERR! node_modules/react
npm ERR!   peer react@"18.1.0" from react-native@0.70.5
npm ERR!   node_modules/react-native
npm ERR!     peer react-native@"^0.70.1" from @nrwl/react-native@15.0.13
npm ERR!     node_modules/@nrwl/react-native
npm ERR!       dev @nrwl/react-native@"^15.0.13" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR!
npm ERR! See /Users/yj/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/yj/.npm/_logs/2022-11-12T19_32_01_683Z-debug-0.log
```

**Comment by xiongemi** [comment]
this issue should be fixed after expo is upgrade 48 with react 18.2 support.
